tests/test_parts.py

```diff
@@ -686,3 +686,76 @@ def test_bulk_create_parts_without_property_value(self):
             self.client._create_parts_bulk(
                 parts=[self.part_dict]
             )
+
+
+@skipIf(not TEST_FLAG_IS_PIM3, reason="This tests is designed for PIM version 3, expected to fail on older PIM3")
+@pytest.mark.skipif("os.getenv('TRAVIS', False) or os.getenv('GITHUB_ACTIONS', False)",
+                    reason="Skipping tests when using Travis or Github Actions, as not Auth can be provided")
+class TestBulkPartsDeletion(TestBetamax):
+    def setUp(self):
+        super().setUp()
+        self.parts = list()
+        self.wheel_model = self.project.model(name="Wheel")
+        self.bike_instance = self.project.part(name="Bike")
+
+        self.diameter_prop = self.wheel_model.property(name="Diameter")
+        self.spokes_prop = self.wheel_model.property(name="Spokes")
+        self.rim_material_prop = self.wheel_model.property(name="Rim Material")
+        self.tire_thickness_prop = self.wheel_model.property(name="Tire Thickness")
+
+        for idx in range(1, 5):
+            part_dict = {
+                "name": "Wheel {}".format(idx),
+                "parent_id": self.bike_instance.id,
+                "description": "",
+                "model_id": self.wheel_model.id,
+                "properties": [
+                    {
+                        "name": self.diameter_prop.name,
+                        "value": float("{}.{}".format(idx, idx)),
+                        "model_id": self.diameter_prop.id,
+                    },
+                    {
+                        "name": self.spokes_prop.name,
+                        "value": "{}".format(idx),
+                        "model_id": self.spokes_prop.id
+                    },
+                    {
+                        "name": self.rim_material_prop.name,
+                        "value": "Material {}".format(idx),
+                        "model_id": self.rim_material_prop.id
+                    },
+                    {
+                        "name": self.tire_thickness_prop.name,
+                        "value": float("{}.{}".format(idx, idx)),
+                        "model_id": self.tire_thickness_prop.id
+                    }
+                ]
+            }
+            self.parts.append(part_dict)
+        self.parts_created = list(self.client._create_parts_bulk(
+            parts=self.parts
+        ))
+
+    def tearDown(self):
+        for part in self.parts_created:
+            # In case the parts have not been deleted and a clean up is required
+            try:
+                part.delete()
+            except APIError:
+                pass
+
+    def test_bulk_delete_parts(self):
+        self.client._delete_parts_bulk(parts=self.parts_created)
+        for idx in range(1, 5):
+            with self.subTest(idx=idx):
+                with self.assertRaises(NotFoundError):
+                    self.project.part(name="Wheel {}".format(idx))
+
+    def test_bulk_delete_parts_with_part_ids(self):
```

Comment:
Both tests share much code. Actually only the `parts` element changes.

Reply:
Agree, but the idea is to make sure the coverage is as high as possible, so I need to test whether it is processed correctly for `Part` objects and `Part` ids. And since I need those parts to delete them in the first place, and those parts are created in the setUp(), then i need 2 different tests.

Reply:
But you can, per test, still use 1 shared function?

Reply:
Oh, I think I understand what you mean, will send a list with 2 parts and 2 uuids and that should reduce the number of duplicated code.